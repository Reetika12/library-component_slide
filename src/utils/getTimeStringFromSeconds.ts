export default function getTimeStringFromSeconds(value: number): string {
	if (typeof value !== 'number') {
		return '';
	}

	let timeText = '';
	let time = value;

	if (value === 0) {
		return '0 Sec';
	}

	if (Math.floor(time / 3600) !== 0) {
		timeText += `${Math.floor(time / 3600)} Hr`;
		time %= 3600;
	}

	if (Math.floor(time / 60) !== 0) {
		if (timeText) {
			timeText += ' ';
		}
		timeText += `${Math.floor(time / 60)} Min`;
		time %= 60;
	}

	if (time) {
		if (timeText) {
			timeText += ' ';
		}
		timeText += `${time} Sec`;
	}

	return timeText;
}
