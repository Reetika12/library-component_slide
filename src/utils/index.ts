export { default as noop } from './noop';
export { default as getTimeStringFromSeconds } from './getTimeStringFromSeconds';
