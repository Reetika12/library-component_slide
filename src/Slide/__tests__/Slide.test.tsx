import * as React from 'react';
// import { act } from 'react-dom/test-utils';
// import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Slide } from '../index';
import { noop } from '../../utils';

describe('Slide', function test() {
	it('should render correctly for cms', () => {
		const component = shallow(
			<Slide
				slideId="asd"
				slideImgUrl="https://www.asiatechdaily.com/wp-content/uploads/2020/01/Byjus-770x433.jpg"
				onSelectClick={noop}
				slideTimer={180}
				slideNumber={1}
				slideName="Live Quiz with large name going out"
			/>
		);

		expect(component).toMatchSnapshot();
	});

	it('should render new slide correctly for cms', () => {
		const component = shallow(
			<Slide
				slideId="asd"
				slideImgUrl="https://www.asiatechdaily.com/wp-content/uploads/2020/01/Byjus-770x433.jpg"
				onSelectClick={noop}
			/>
		);

		expect(component).toMatchSnapshot();
	});
});
