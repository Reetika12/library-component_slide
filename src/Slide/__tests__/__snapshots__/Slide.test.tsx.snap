// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should render correctly for cms 1`] = `
<div
  className="slide "
  onClick={[Function]}
  style={
    Object {
      "height": " 124px",
      "width": "176px",
    }
  }
>
  <div
    className="slide__img_box"
  >
    <div
      className="slide__img"
    >
      <img
        alt="slide"
        src="https://www.asiatechdaily.com/wp-content/uploads/2020/01/Byjus-770x433.jpg"
      />
    </div>
  </div>
  <div
    className="slide__content_box"
  >
    <div
      className="slide__slide_number"
    >
      01
    </div>
    <div
      className="slide__slide_name"
    >
      Live Quiz with large name going out
    </div>
    <div
      className="slide__slide_left_content"
    >
      <div
        className="slide__slide_timer"
      >
        3 Min
      </div>
    </div>
  </div>
</div>
`;

exports[`Slide should render new slide correctly for cms 1`] = `
<div
  className="slide "
  onClick={[Function]}
  style={
    Object {
      "height": " 124px",
      "width": "176px",
    }
  }
>
  <div
    className="slide__img_box"
  >
    <div
      className="slide__img"
    >
      <img
        alt="slide"
        src="https://www.asiatechdaily.com/wp-content/uploads/2020/01/Byjus-770x433.jpg"
      />
    </div>
  </div>
  <div
    className="slide__content_box"
  >
    <div
      className="slide__slide_name"
    >
      New Slide
    </div>
    <div
      className="slide__slide_left_content"
    />
  </div>
</div>
`;
